examples/joystick/Joystick.cpp

```diff
@@ -0,0 +1,176 @@
+        sstr << "Joystick " << index << ":";
+        texts["ID"].first.setString(sstr.str());
+        texts["ID"].second.setString(sf::Joystick::getIdentification(index).name);
+    }
+
+    // Update joystick axes
+    void updateAxes(unsigned int index)
+    {
+        for (unsigned int j = 0; j < sf::Joystick::AxisCount; ++j)
+        {
+            if (sf::Joystick::hasAxis(index, static_cast<sf::Joystick::Axis>(j)))
+                set(axislabels[j], sf::Joystick::getAxisPosition(index, static_cast<sf::Joystick::Axis>(j)));
+        }
+    }
+
+    // Update joystick buttons
+    void updateButtons(unsigned int index)
+    {
+        for (unsigned int j = 0; j < sf::Joystick::getButtonCount(index); ++j)
+        {
+            sstr.str("");
+            sstr << "Button " << j;
+
+            set(sstr.str().c_str(), sf::Joystick::isButtonPressed(index, j));
+        }
+    }
+
+    // Helper to update displayed joystick values
+    void updateValues(unsigned int index)
+    {
+        if (sf::Joystick::isConnected(index)) {
+            // Update the label-value sf::Text objects based on the current joystick state
+            updateIdentification(index);
+            updateAxes(index);
+            updateButtons(index);
+        }
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+/// Entry point of application
+///
+/// \return Application exit code
+///
+////////////////////////////////////////////////////////////
+int main()
+{
+    // Create the window of the application
+    sf::RenderWindow window(sf::VideoMode(480, 640), "Joystick", sf::Style::Close);
+    window.setVerticalSyncEnabled(true);
+
+    // Load the text font
+    sf::Font font;
+    if (!font.loadFromFile("resources/sansation.ttf"))
+        return EXIT_FAILURE;
+
+    // Set up our string conversion parameters
+    sstr.precision(2);
+    sstr.setf(std::ios::fixed | std::ios::boolalpha);
+
+    // Set up our label-value sf::Text objects
+    texts["ID"].first.setPosition(5.f, 5.f);
+    texts["ID"].second.setPosition(80.f, 5.f);
+
+    for (unsigned int i = 0; i < sf::Joystick::AxisCount; ++i)
+    {
+        std::pair<sf::Text, sf::Text>& text = texts[axislabels[i]];
+
+        text.first.setPosition(5.f, 5.f + ((i + 2) * font.getLineSpacing(14)));
+        text.first.setString(std::string(axislabels[i]) + ":");
+
+        text.second.setPosition(80.f, 5.f + ((i + 2) * font.getLineSpacing(14)));
+    }
+
+    for (unsigned int i = 0; i < sf::Joystick::ButtonCount; ++i)
+    {
+        sstr.str("");
+        sstr << "Button " << i;
+        std::pair<sf::Text, sf::Text>& text = texts[sstr.str()];
+
+        text.first.setPosition(5.f, 5.f + ((sf::Joystick::AxisCount + i + 2) * font.getLineSpacing(14)));
+        text.first.setString(sstr.str() + ":");
+
+        text.second.setPosition(80.f, 5.f + ((sf::Joystick::AxisCount + i + 2) * font.getLineSpacing(14)));
+    }
+
+    for (Texts::iterator i = texts.begin(); i != texts.end(); ++i)
```

Comment:
Also, maybe just me, but I prefer to reserve `i`, `j`, etc. for numerical indices, while using `a`, `b`, etc. for iterators (even though [i]terator makes sense).

Reply:
My personal convention, and thus most likely what's used in SFML, is to use `it` for iterators.